### DISCUSSION TITLE
Sending C# console output to another profile/tab with cmd.exe

### DISCUSSION THREAD

**Discussion by codenotworking** [microsoft/terminal#18146] [Q&A] [answered]
Within a C# console application, being executed within Windows Terminal, I'm trying to have it create/spawn a new cmd.exe within Windows Terminal, and redirect some output to it, but it's not working.

Sample C# code:

```csharp
ProcessStartInfo psi = new() {
    FileName = "wt.exe",
    Arguments = "--window 0 new-tab cmd.exe /k",
    RedirectStandardError = true,
    RedirectStandardInput = true,
    RedirectStandardOutput = true,
    UseShellExecute = false
};

using Process process = Process.Start(psi)!;

using StreamReader reader = process!.StandardOutput;
using StreamWriter writer = process!.StandardInput;

writer.WriteLine("Hello from new console");
```

Is this something possible with Windows Terminal?  Any input would be appreciated.

**Comment by DHowett** [comment]
Sorry, just so I understand - you're trying to _launch the Command Prompt_ (which will produce its own output) and then _also write more unrelated output to the same window that is hosting it_ which will conflict with the output that CMD produces?

**Reply by DHowett** [reply]
The end result being that it should look like this?

```
Microsoft Windows [Version 10.0.26310.5000]
(c) Microsoft Corporation. All rights reserved.

C:\Users\duhowett>Hello from new console
```

**Comment by codenotworking** [comment]
Initial cmd.exe where I run the console app from:  
![image](https://github.com/user-attachments/assets/fd06a467-5d0a-497e-8fde-a75fd2eb6619)

New cmd.exe opened which I want to redirect output to:
![image](https://github.com/user-attachments/assets/889b188b-d235-4af1-b277-340b8416868b)

Is it possible to redirect output from one console to the other in Windows Terminal?

**Comment by codenotworking** [comment, marked as answer]
I found a solution using [Memory-mapped files](https://learn.microsoft.com/en-us/dotnet/standard/io/memory-mapped-files).  I just have to create two console applications.
